src/DotVVM.Framework/Resources/Scripts/metadata/metadataHelper.ts

```diff
@@ -0,0 +1,40 @@
+﻿import { getTypeInfo } from "./typeMap";
+
+export function getTypeId(viewModel: any): string {
+    if (typeof viewModel !== "object") {
+        throw Error("Expected object but received: " + typeof viewModel);
+    }
+
+    return ko.unwrap(viewModel.$type);
+}
+
+export function getTypeMetadata(viewModel: any): TypeMetadata {
```

Comment:
It would be better to accept a type id to be consistent with `getEnumMetadata`. If you think it's helpful to also accept an object, we can accept both - `viewModel: string | object`.